pkg/services/serviceaccounts/database/store.go

```diff
@@ -422,6 +419,16 @@ func (s *ServiceAccountsStoreImpl) CreateServiceAccountFromApikey(ctx context.Co
 		}
 
 		if err := s.assignApiKeyToServiceAccount(sess, key.ID, newSA.ID); err != nil {
+			// we have to revert the created service account
+			// because currently we are not sharing the same transaction
+			// if we would share the same transaction, we could just rollback
+			// this ensures that a retry of the migration will not create duplicate service accounts
+			// and therefore fail due to "already created service account"
+			err := s.userService.Delete(ctx, &user.DeleteUserCommand{UserID: newSA.ID})
```

Comment:
we are not sharing the same transaction in these two calls.

1. creating service account is not rolled back if the assignment of apikey to service account fails
2. we need to therefore delete the created service account, in case the transaction retries again. otherwise we hit a "already created service account"